system/lib/dlmalloc.c

```diff
@@ -850,6 +850,39 @@ extern "C" {
     /* ------------------- Declarations of public routines ------------------- */
     
 #ifndef USE_DL_PREFIX
+// XXX Emscripten XXX
```

Comment:
making all these symbols weak may have side effects, are we not worried about that?

Reply:
They were already weak to begin with. The difference is that now, on wasm backend, we use aliases instead of renaming all the functions.